### PR TITLE
Fixed part 2 of issue3080: UI suspension button for CronJob

### PR DESCRIPTION
**Pull request by EdanSneh**
Creates new button for suspending cronjob, with a component and a controller. New suspend API handler in backend, with a suspension library.

### CODE REVIEW

File: src/app/frontend/common/components/resourcecard/resourcecardsuspendmenuitem_component.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          },
+        });
+  }
+
+  successdisable_() {
+    this.log_.info(`Successfully updated cronjob suspension to true`);
+    this.state_.reload();
+  }
+
+  successenable_() {
```
Comment (maciaszczykm) [inline src/app/frontend/common/components/resourcecard/resourcecardsuspendmenuitem_component.js]: Use camelCase.

File: src/app/frontend/common/components/resourcecard/resourcecardsuspendmenuitem_component.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            method: 'PUT',
+            transformRequest: function(headers) {
+              headers = angular.extend({}, headers, {'Content-Type': 'application/json'});
+              return angular.toJson(headers);
+            },
+          },
+        });
+  }
+
+  successdisable_() {
```
Comment (maciaszczykm) [inline src/app/frontend/common/components/resourcecard/resourcecardsuspendmenuitem_component.js]: Use camelCase.

File: src/app/backend/suspend/suspender.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+func SuspendCronJob (client client.Interface, namespace, name, suspend string) (cj *SuspendStatus, err error) {
+  cj = new(SuspendStatus)
+  cron, err := client.BatchV1beta1().CronJobs(namespace).Get(name, meta.GetOptions{})
+  if err != nil {
+    return nil, err
+  }
+
+  if suspend == "true" {
+  *cron.Spec.Suspend = true
```
Comment (maciaszczykm) [inline src/app/backend/suspend/suspender.go]: Run code formatting.

**Comment by EdanSneh** [comment]
@maciaszczykm I implemented the changes you requested. Not sure why travis is failing now
@austbot I added specs, but I still don't think it will fulfill codecov

File: src/app/backend/suspend/suspender.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func SuspendCronJob(client client.Interface, namespace, name, suspend string) (cj *SuspendStatus, err error) {
+	cj = new(SuspendStatus)
+	cron, err := client.BatchV1beta1().CronJobs(namespace).Get(name, meta.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if suspend == "true" {
```
Comment (maciaszczykm) [inline src/app/backend/suspend/suspender.go]: Why suspend is not a boolean?

**Comment by maciaszczykm** [comment]
I am closing it as we have just finished migration and it is stale. Please reopen if you would like to finish it.

/close

**Pull request closed without merging**